altair/vegalite/v5/api.py

```diff
@@ -786,15 +799,33 @@ def binding_range(**kwargs):
     return core.BindRange(input="range", **kwargs)
 
 
+_TSchemaBase = typing.TypeVar("_TSchemaBase", bound=core.SchemaBase)
+
+
+@typing.overload
```

Comment:
These overloads originate from https://github.com/dangotbanned/altair/blob/e159ea08f970e3fd388044cf07cb670987f6f967/altair/vegalite/v5/api.py#L1479

They fully encpasualte the complex internal logic of `condition`, which on this branch is now required due to `Chart.encode` being understood by type checkers.

A very simple example of `encode` not being understood, is the correction required from `Color` -> `Fill` here https://github.com/dangotbanned/altair/blob/e4ab7052e9a1e62ff1fd80379864489c69a1e020/tests/examples_arguments_syntax/hexbins.py#L33

Relating back to `condition`, this message:
> Re-running `mypy` output 'Found 63 errors in 47 files (checked 360 source files)', tests/examples

Almost all of these were `condition`, which are now **not** emitting errors - entirely due to the new overloads. I didn't need to add any "# type: ignore" comments in the examples, which would have been a less desirable alternative.